ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -1440,64 +1441,112 @@ template<class FluidSystem,class Scalar>
     std::sort(globalFailedCellsPd.begin(), globalFailedCellsPd.end());
 
+    auto ijkString = [this](const std::size_t globalIndex)
+    {
+        const auto ijk = this->eclState_.gridDims().getIJK(globalIndex);
+
+        return fmt::format("({},{},{})", ijk[0] + 1, ijk[1] + 1, ijk[2] + 1);
+    };
+
+    const auto maxNumCellsFaillog = static_cast<std::size_t>(20);
+
     if (!globalFailedCellsPb.empty()) {
+        const auto u = std::unique(globalFailedCellsPb.begin(),
+                                   globalFailedCellsPb.end());
+
+        const auto numFailed = static_cast<std::size_t>
+            (std::distance(globalFailedCellsPb.begin(), u));
+
         std::stringstream errlog;
-        errlog << "Finding the bubble point pressure failed for " << globalFailedCellsPb.size() << " cells [";
-        errlog << globalFailedCellsPb[0];
-        const size_t maxElems = std::min(maxNumCellsFaillog, globalFailedCellsPb.size());
-        for (size_t i = 1; i < maxElems; ++i) {
-            errlog << ", " << globalFailedCellsPb[i];
+        errlog << "Finding the bubble point pressure failed for "
+               << numFailed << " cell"
+               << ((numFailed != std::size_t{1}) ? "s" : "")
+               << " [" << ijkString(globalFailedCellsPb[0]);
+
+        const auto maxElems = std::min(maxNumCellsFaillog, numFailed);
+        for (auto i = 1 + 0*maxNumCellsFaillog; i < maxElems; ++i) {
+            errlog << ", " << ijkString(globalFailedCellsPb[i]);
         }
-        if (globalFailedCellsPb.size() > maxNumCellsFaillog) {
+
+        if (numFailed > maxNumCellsFaillog) {
             errlog << ", ...";
         }
-        errlog << "]";
+
+        errlog << ']';
+
         OpmLog::warning("Bubble point numerical problem", errlog.str());
     }
+
     if (!globalFailedCellsPd.empty()) {
+        const auto u = std::unique(globalFailedCellsPd.begin(),
+                                   globalFailedCellsPd.end());
+
+        const auto numFailed = static_cast<std::size_t>
+            (std::distance(globalFailedCellsPd.begin(), u));
+
         std::stringstream errlog;
-        errlog << "Finding the dew point pressure failed for " << globalFailedCellsPd.size() << " cells [";
-        errlog << globalFailedCellsPd[0];
-        const size_t maxElems = std::min(maxNumCellsFaillog, globalFailedCellsPd.size());
-        for (size_t i = 1; i < maxElems; ++i) {
-            errlog << ", " << globalFailedCellsPd[i];
+        errlog << "Finding the dew point pressure failed for "
+               << numFailed << " cell"
+               << ((numFailed != std::size_t{1}) ? "s" : "")
+               << " [" << ijkString(globalFailedCellsPd[0]);
+
+        const auto maxElems = std::min(maxNumCellsFaillog, numFailed);
+        for (auto i = 1 + 0*maxNumCellsFaillog; i < maxElems; ++i) {
+            errlog << ", " << ijkString(globalFailedCellsPd[i]);
         }
-        if (globalFailedCellsPd.size() > maxNumCellsFaillog) {
+
```

Comment:
Isn't this the very same algorithm as above for globalFailedCellsPb? Why not factor that out into a method and call that twice to reduce code duplication?

Reply:
> Why not factor that out into a method and call that twice to reduce code duplication?

I'll do that but I won't add a new member function.